File: Practice_1 (C#)/Practice_1/UserInterface.cs
using System;
using ListLib;

namespace Practice_1
{
    internal class UserInterface
    {
        // Метод, який запускає користувацький інтерфейс.
        public static void Start(ref DoublyLinkedList<Student> students)
        {
            // Перший вивід всіх можливих команд.
            CommandsHandler.CommandHelp();
            Console.WriteLine();
            string input;

            while (true)
            {
                // Очікування введення команди.
                Console.Write(" ");
                input = Console.ReadLine().ToLower();

                // Перевірка на співпадіння з можливими командами.
                if (input == "/addfirst")
                {
                    CommandsHandler.CommandAddFirst(ref students);
                    Console.WriteLine();
                    continue;
                }
                if (input == "/clear")
                {
                    Console.Clear();
                    continue;
                }
                if (input == "/end")
                {
                    break;
                }
                if (input == "/help")
                {
                    CommandsHandler.CommandHelp();
                    Console.WriteLine();
                    continue;
                }
                if (input == "/length")
                {
                    CommandsHandler.CommandLenght(ref students);
                    Console.WriteLine();
                    continue;
                }
                if (input == "/removelast")
                {
                    CommandsHandler.CommandRemoveLast(ref students);
                    Console.WriteLine();
                    continue;
                }
                if (input == "/search")
                {
                    CommandsHandler.CommandSearch(ref students);
                    Console.WriteLine();
                    continue;
                }
                if (input == "/show")
                {
                    CommandsHandler.CommandShow(ref students);
                    Console.WriteLine();
                    continue;
                }
                if (input == "/showback")
                {
                    CommandsHandler.CommandShowBack(ref students);
                    Console.WriteLine();
                    continue;
                }
                if (input == "/sort")
                {
                    CommandsHandler.CommandSort(ref students);
                    Console.WriteLine();
                    continue;
                }

                // Якщо введена команда не співпадає з можливими, прохання ввести команду ще раз.
                Console.Clear();
                Console.WriteLine(" Команда ввведена невірно. Спробуйте ще раз.");
                CommandsHandler.CommandHelp();
            }
        }
    }
}

File: Practice_2 (Java)/src/UserInterface.java
import java.io.IOException;
import java.util.Scanner;
import ua.KarpovIvan.atd.*;

public class UserInterface
{
    /*
     * Метод, який запускає користувацький інтерфейс.
     */
    public static void start(DoublyLinkedList<Student> students) throws IOException, InterruptedException {
        var in = new Scanner(System.in); /* Для зчитування введеня з консолі */
        String input;

        /* Перший вивід всіх можливих команд. */
        CommandsHandler.commandHelp();
        System.out.println();

        while (true) {
            /* Очікування введення команди. */
            System.out.print(" ");
            input = in.nextLine();

            /* Перевірка на співпадіння з можливими командами. */
            if (input.equals("/addfirst")) {
                System.out.print("\r");
                CommandsHandler.commandAddFirst(students);
                System.out.println();
                continue;
            }
            if (input.equals("/clear")) {
                new ProcessBuilder("cmd", "/c", "cls").inheritIO().start().waitFor();
                continue;
            }
            if (input.equals("/end")) {
                break;
            }
            if (input.equals("/help")) {
                CommandsHandler.commandHelp();
                System.out.println();
                continue;
            }
            if (input.equals("/length")) {
                CommandsHandler.commandLength(students);
                System.out.println();
                continue;
            }
            if (input.equals("/removelast")) {
                CommandsHandler.commandRemoveLast(students);
                System.out.println();
                continue;
            }
            if (input.equals("/search")) {
                CommandsHandler.commandSearch(students);
                System.out.println();
                continue;
            }
            if (input.equals("/show")) {
                CommandsHandler.commandShow(students);
                System.out.println();
                continue;
            }
            if (input.equals("/showback")) {
                CommandsHandler.commandShowBack(students);
                System.out.println();
                continue;
            }
            if (input.equals("/sort")) {
                CommandsHandler.commandSort(students);
                System.out.println();
                continue;
            }

            /* Якщо введена команда не співпадає з можливими, прохання ввести команду ще раз. */
            System.out.println(" Команда ввведена невiрно. Спробуйте ще раз.");
            CommandsHandler.commandHelp();
            System.out.println();
        }
    }
}